### DISCUSSION TITLE
avoid_negative_ts -  why change default recommendation? has implementation changed?

### DISCUSSION THREAD

**Discussion by bryc** [mifi/lossless-cut#2265] [Q&A] [answered]
Making this a Q&A discussion as it is probably not a bug.
I've just tried the latest LC and noticed that I now get a warning when attempting to merge segments of a x264/aac mp4.

LC recommends `make_non_negative`. 

But here's the kicker - the old version I use, actually gives a warning if `make_non_negative` is used. I've always used `make_zero`, which is the default for the older version. I never needed to change it; it's never given me an issue (as far as I know?).

So, this causes some confusion.

Has there been an implementation change? is `make_zero` safer on older versions that later became unsafe on newer versions?

Or are the behaviours the same between versions, and this is just purely a guidance change?

Personally, I've noticed that using `make_non_negative` can have inconsistent behavior between video players. For example If you cut 3-5 frames ahead of a key frame, This option appears to allow ffplay (and Chromium) to skip past those first 3-5 frames somehow, but VLC (and Firefox) does not. I thought it'd be useful as an alternative to "smart cut", but it's a problem if it doesn't work in VLC/Firefox. `make_zero` on the other hand, has consistent behavior between most players, when I cut 3 frames ahead, it always snaps to the previous keyframe.

So I'd rather keep using `make_zero` if the warning is benign.

**Comment by mifi** [comment, marked as answer]
It’s purely a recommendation change. It was made here: https://github.com/mifi/lossless-cut/commit/dca83a8ab42450dbd7ff93709020955d044854e5
Here’s the reasoning behind the change: https://github.com/mifi/lossless-cut/discussions/1874#discussioncomment-8268873
